src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -416,21 +416,20 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     // the end of the current batch.
     const completionCallbacks = root.completionCallbacks;
     if (completionCallbacks !== null) {
-      processUpdateQueue(completionCallbacks, null, null, null, completedAt);
-      const callbackList = completionCallbacks.callbackList;
-      if (callbackList !== null) {
-        // Add new callbacks to list of completion callbacks
+      processUpdateQueue(completionCallbacks, null, null, completedAt);
+      // Add new callbacks to list of completion callbacks
+      let callbackNode = completionCallbacks.firstCallback;
```

Comment:
TODO: Should remove callback from update object here so that it isn't called twice. Add test.